cmd/gaia/app/export.go

```diff
@@ -56,34 +56,13 @@ func (app *GaiaApp) ExportAppStateAndValidators(forZeroHeight bool) (
 // prepare for fresh start at zero height
 func (app *GaiaApp) prepForZeroHeightGenesis(ctx sdk.Context) {
 
-	/* TODO XXX check some invariants */
-
-	height := ctx.BlockHeight()
-
-	valAccum := sdk.ZeroDec()
-	vdiIter := func(_ int64, vdi distr.ValidatorDistInfo) bool {
-		lastValPower := app.stakeKeeper.GetLastValidatorPower(ctx, vdi.OperatorAddr)
-		valAccum = valAccum.Add(vdi.GetValAccum(height, sdk.NewDecFromInt(lastValPower)))
-		return false
-	}
-	app.distrKeeper.IterateValidatorDistInfos(ctx, vdiIter)
-
-	lastTotalPower := sdk.NewDecFromInt(app.stakeKeeper.GetLastTotalPower(ctx))
-	totalAccum := app.distrKeeper.GetFeePool(ctx).GetTotalValAccum(height, lastTotalPower)
-
-	if !totalAccum.Equal(valAccum) {
-		panic(fmt.Errorf("validator accum invariance: \n\tfee pool totalAccum: %v"+
-			"\n\tvalidator accum \t%v\n", totalAccum.String(), valAccum.String()))
-	}
-
-	fmt.Printf("accum invariant ok!\n")
-
-	/* END TODO XXX */
+	/* Just to be safe, assert the invariants on current state. */
+	app.assertRuntimeInvariantsWith(ctx)
```

Comment:
confused by this name "With.."

Reply:
Changed to `assertRuntimeInvariantsOnContext`.